File: 114/114_version1.py
class Solution:
    def flatten(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if root:
            self.flatten(root.left)
            temp = root.right
            root.right = root.left
            self.flatten(temp)
            pointer = root
            while pointer.right:
                pointer = pointer.right
            pointer.right = temp
            root.left = None

File: 114/114_version1.java
class Solution {
    public void flatten(TreeNode root) {
        if(root!=null){
            flatten(root.left);
            TreeNode temp = root.right;
            root.right = root.left;
            root.left = null;
            TreeNode pointer = root;
            while(pointer.right!=null) pointer = pointer.right;
            flatten(temp);
            pointer.right = temp;
        }
    }
}
